File: src/AutomatonTheory/JGraph/DialogBoxes/AddTransitionDialog.java
package AutomatonTheory.JGraph.DialogBoxes;

import javax.swing.*;

public class AddTransitionDialog {

    public String[] States;
    public String[] Alphabet;

    public String originState;
    public String destinyState;
    public String symbol;

    public JComboBox<String> orginStateComboBox = new JComboBox<String>();
    public JComboBox<String> detinyStateComboBox = new JComboBox<String>();
    public JComboBox<String> symbolComboBox = new JComboBox<String>();


    public AddTransitionDialog(String[] states, String[] alphabet){
        States = states;
        Alphabet = alphabet;
    }

    public void displayGUI() {
        JOptionPane.showConfirmDialog(null,
                                getPanel(),
                                "AddTransition",
                                JOptionPane.OK_CANCEL_OPTION,
                                JOptionPane.PLAIN_MESSAGE);

        originState = orginStateComboBox.getSelectedItem().toString();
        destinyState = detinyStateComboBox.getSelectedItem().toString();
        symbol = symbolComboBox.getSelectedItem().toString();

    }

    private JPanel getPanel(){

        JPanel panel = new JPanel();
        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
        JLabel originLabel = new JLabel("Choose Origin: ");
        orginStateComboBox = new JComboBox<String>(States);
        JLabel destinyLabel = new JLabel("Choose Destiny: ");
        detinyStateComboBox = new JComboBox<String>(States);
        JLabel symbolLabel = new JLabel("Choose Symbol: ");
        symbolComboBox = new JComboBox<String>(Alphabet);

        panel.add(originLabel);
        panel.add(orginStateComboBox);
        panel.add(destinyLabel);
        panel.add(detinyStateComboBox);
        panel.add(symbolLabel);
        panel.add(symbolComboBox);

        return panel;
    }
    public static void main(String[] args){
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                new AddTransitionDialog(new String[]{"q0","q1"}, new String[]{"0","1"}).displayGUI();
            }
        });
    }

}

File: src/AutomatonTheory/Kotlin/Swing/DialogBoxes/AddTransitionDialog.kt
package AutomatonTheory.Kotlin.Swing.DialogBoxes

import javax.swing.*

class AddTransitionDialog(var States: Array<String>, var Alphabet: Array<String>) {

    var originState: String = ""
    var destinyState: String = ""
    var symbol: String = ""

    var orginStateComboBox = JComboBox<String>()
    var detinyStateComboBox = JComboBox<String>()
    var symbolComboBox = JComboBox<String>()

    var valor: Int = 0

    fun displayGUI() {
        valor = JOptionPane.showConfirmDialog(null,
                panel,
                "AddTransition",
                JOptionPane.OK_CANCEL_OPTION,
                JOptionPane.PLAIN_MESSAGE)

        originState = orginStateComboBox.selectedItem.toString()
        destinyState = detinyStateComboBox.selectedItem.toString()
        symbol = symbolComboBox.selectedItem.toString()

    }

    private val panel: JPanel
        get() {

            val panel = JPanel()
            panel.layout = BoxLayout(panel, BoxLayout.Y_AXIS)
            val originLabel = JLabel("Choose Origin: ")
            orginStateComboBox = JComboBox(States)
            val destinyLabel = JLabel("Choose Destiny: ")
            detinyStateComboBox = JComboBox(States)
            val symbolLabel = JLabel("Choose Symbol: ")
            symbolComboBox = JComboBox(Alphabet)

            panel.add(originLabel)
            panel.add(orginStateComboBox)
            panel.add(destinyLabel)
            panel.add(detinyStateComboBox)
            panel.add(symbolLabel)
            panel.add(symbolComboBox)

            return panel
        }

    companion object {
        @JvmStatic fun main(args: Array<String>) {
            SwingUtilities.invokeLater { AddTransitionDialog(arrayOf("q0", "q1"), arrayOf("0", "1")).displayGUI() }
        }
    }

}
